src/engine/strat_engine/thindev.rs

```diff
@@ -68,11 +77,15 @@ impl ThinDev {
 
     }
 
-    pub fn teardown(&mut self, dm: &DM) -> EngineResult<()> {
+    pub fn teardown(&self, dm: &DM) -> EngineResult<()> {
         try!(dm.device_remove(&DevId::Name(&self.name), DmFlags::empty()));
         Ok(())
     }
 
+    pub fn name(&self) -> &str {
+        self.dev_info.name().clone()
```

Comment:
Ok.  I was thinking it would be best to not return references to strings.  I think it is like returning a pointer to a string inside a C structure - typically not a good idea for an API in C.  But, I think you are indicating  rust lifetimes will protect us... (?)